Introduction

### How to cite IQ-TREE?

> To maintain IQ-TREE, support users and secure fundings, it is important for us that 
> you cite the following papers, whenever the corresponding features were applied for your analysis.**
>
> Example 1: We obtained branch supports with the ultrafast bootstrap (Hoang et al., 2018) 
> implemented in the IQ-TREE 2 software (Minh et al., 2020).
>
> Example 2: We used IQ-TREE 2 (Minh et al., 2020) to infer the maximum-likelihood tree using the edge-linked 
> partition model (Chernomor et al., 2016).

General citation for IQ-TREE 2:

* B.Q. Minh, H.A. Schmidt, O. Chernomor, D. Schrempf, M.D. Woodhams, A. von Haeseler, R. Lanfear (2020) 
  IQ-TREE 2: New models and efficient methods for phylogenetic inference in the genomic era.
  *Mol. Biol. Evol.*, 37:1530-1534. <https://doi.org/10.1093/molbev/msaa015>

When using tree mixture models (MAST) please cite:

* T.K.F. Wong, C. Cherryh, A.G. Rodrigo, M.W. Hahn, B.Q. Minh, R. Lanfear (2024)
  MAST: Phylogenetic Inference with Mixtures Across Sites and Trees.
  _Syst. Biol._, in press. <https://doi.org/10.1093/sysbio/syae008>

When computing concordance factors please cite:

* Y.K. Mo, R. Lanfear, M.W. Hahn, B.Q. Minh (2023)
  Updated site concordance factors minimize effects of homoplasy and taxon sampling.
  _Bioinformatics_, 39:btac741. <https://doi.org/10.1093/bioinformatics/btac741>

When using AliSim to simulate alignments please cite:

* N. Ly-Trong, G.M.J. Barca, B.Q. Minh (2023)
  AliSim-HPC: parallel sequence simulator for phylogenetics.
  *Bioinformatics*, 39:btad540. <https://doi.org/10.1093/bioinformatics/btad540>

When estimating amino-acid Q matrix please cite:

* B.Q. Minh, C. Cao Dang, L.S. Vinh, R. Lanfear (2021)
  QMaker: Fast and accurate method to estimate empirical models of protein evolution.
  _Syst. Biol._, 70:1046–1060. <https://doi.org/10.1093/sysbio/syab010>

When using the heterotachy GHOST model "+H" please cite:

* S.M. Crotty, B.Q. Minh, N.G. Bean, B.R. Holland, J. Tuke, L.S. Jermiin, A. von Haeseler (2020)
  GHOST: Recovering Historical Signal from Heterotachously Evolved Sequence Alignments.
  _Syst. Biol._, 69:249-264. <https://doi.org/10.1093/sysbio/syz051>

When using the tests of symmetry please cite:

* S. Naser-Khdour, B.Q. Minh, W. Zhang, E.A. Stone, R. Lanfear (2019) 
  The Prevalence and Impact of Model Violations in Phylogenetic Analysis. 
  *Genome Biol. Evol.*, 11:3341-3352. <https://doi.org/10.1093/gbe/evz193>

When using polymorphism-aware models please cite:

* D. Schrempf, B.Q. Minh, A. von Haeseler, C. Kosiol (2019) 
  Polymorphism-aware species trees with advanced mutation models, bootstrap, and rate heterogeneity. 
  *Mol. Biol. Evol.*, 36:1294–1301. <https://doi.org/10.1093/molbev/msz043>

For the ultrafast bootstrap (UFBoot) please cite:

* D.T. Hoang, O. Chernomor, A. von Haeseler, B.Q. Minh, and L.S. Vinh (2018) 
  UFBoot2: Improving the ultrafast bootstrap approximation. 
  *Mol. Biol. Evol.*, 35:518–522. <https://doi.org/10.1093/molbev/msx281>

When using posterior mean site frequency model (PMSF) please cite:

* H.C. Wang, B.Q. Minh, S. Susko, A.J. Roger (2018) 
  Modeling site heterogeneity with posterior mean site frequency profiles 
  accelerates accurate phylogenomic estimation. 
  *Syst. Biol.*, 67:216–235. <https://doi.org/10.1093/sysbio/syx068>

When using ModelFinder please cite:

* S. Kalyaanamoorthy, B.Q. Minh, T.K.F. Wong, A. von Haeseler, L.S. Jermiin (2017) 
  ModelFinder: Fast model selection for accurate phylogenetic estimates. 
  *Nat. Methods*, 14:587-589. <https://doi.org/10.1038/nmeth.4285>

When using partition models please cite:

* O. Chernomor, A. von Haeseler, B.Q. Minh (2016) 
  Terrace aware data structure for phylogenomic inference from supermatrices. 
  *Syst. Biol.*, 65:997-1008. <https://doi.org/10.1093/sysbio/syw037>

When using IQ-TREE web server please cite:

* J. Trifinopoulos, L.-T. Nguyen, A. von Haeseler, B.Q. Minh (2016) 
  W-IQ-TREE: a fast online phylogenetic tool for maximum likelihood analysis.
  *Nucleic Acids Res.*, 44:W232-W235. <https://doi.org/10.1093/nar/gkw256>

When using IQ-TREE version 1 please cite:

* L. Nguyen, H.A. Schmidt, A. von Haeseler, B.Q. Minh (2015)
  IQ-TREE: A Fast and Effective Stochastic Algorithm for Estimating Maximum-Likelihood Phylogenies.
  _Mol. Biol. and Evol._, 32:268-274. <https://doi.org/10.1093/molbev/msu300>